### PR TITLE
OCPBUGS-42004: Set the Managed Identity client ID

### PR DESCRIPTION
**Pull request by bryan-cox**
Set the Managed Identity client ID

### CODE REVIEW

File: pkg/cloudprovider/azure.go [hunk 1]
```diff
@@ -586,13 +586,14 @@ func (a *Azure) getAuthorizer(env azureapi.Environment, cfg *azureCredentialsCon
...
-	msi := os.Getenv("AZURE_MSI_AUTHENTICATION")
-	if msi == "true" {
+	// Managed Identity Override for ARO HCP
+	managedIdentityClientID := os.Getenv("ARO_HCP_MI_CLIENT_ID")
+	if managedIdentityClientID != "" {
 		options := azidentity.ManagedIdentityCredentialOptions{
 			ClientOptions: azcore.ClientOptions{
 				Cloud: cloudConfig,
 			},
+			ID: azidentity.ClientID(managedIdentityClientID),
```
Comment (fxierh) [inline pkg/cloudprovider/azure.go:596]: The mock-http demo seems to work without specifying the ID. Can this field be omitted?

Comment (bryan-cox) [reply pkg/cloudprovider/azure.go:596]: Someone I talked to from Microsoft said we would need a valid Client ID here when we are using the adapter sidecar containers. More info on that here - https://github.com/openshift/enhancements/pull/1659#issuecomment-2350008814.

**Comment by bryan-cox** [comment]
/hold

Microsoft is requesting to not use the sidecar containers.

**Comment by bryan-cox** [comment]
/close

An official decision was made on Monday to not use sidecars to authenticate with Azure. Closing out this PR.

**Pull request closed without merging**